### ISSUE TITLE
Note editor: Annotation fragment insertion anomalies

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#2556]
https://forums.zotero.org/discussion/95121/bug-with-hiding-and-showing-annonation-citations-and-custom-template

Not the reported bug, but if I set the first template and use Add to Note on an annotation, I get this error:

```
RangeError: Position 283 out of range

    resolve@resource://zotero/note-editor/editor.js:1304:55
    resolveCached@resource://zotero/note-editor/editor.js:1327:48
    resolve@resource://zotero/note-editor/editor.js:1596:58
    insertHTML/<@resource://zotero/note-editor/editor.js:37306:58
    insertHTML@resource://zotero/note-editor/editor.js:42072:5
    index_zotero_EditorInstance/<@resource://zotero/note-editor/editor.js:42231:13
```

It goes away if I reset the template.

(I can reproduce the problem with the second example too.)

**Comment by mrtcode** [comment]
This specific issue is more or less fixed in zotero/note-editor@55b26f9a88944a684d2ecce7b54ffab4c52454f4, but there are random anomalies (annotation or citation can be inserted in random position) when inserting generated annotation fragments into a note. And it might be time consuming to figure what is causing this, because I need to consistently reproduce this and look into ProseMirror core what is happening.

**Comment by mrtcode** [comment]
https://forums.zotero.org/discussion/96932/when-i-add-item-note-the-notes-can-not-show-on-page
